src/js/Content/Features/Community/Inventory/FInventoryMarketHelper.js

```diff
@@ -152,70 +152,88 @@ export default class FInventoryMarketHelper extends Feature {
         }
     }
 
-    _setBackgroundOption(thisItem, assetId, itemActions) {
+    async _setBackgroundOption(thisItem, itemActions) {
 
-        if (!document.querySelector(".inventory_links")) { return; }
         if (itemActions.querySelector(".es_set_background")) { return; }
 
+        // Make sure the selected item is a background
         const viewFullBtn = itemActions.querySelector("a");
-        if (!viewFullBtn) { return; }
+        if (!viewFullBtn || !/public\/images\/items/.test(viewFullBtn.href)) { return; }
 
-        if (!/public\/images\/items/.test(viewFullBtn.href)) { return; }
+        // Get owned backgrounds and the communityitemid for equipped background
+        if (!this.profileBgsOwned) {
+
+            // Retrieve access token from user's profile to use in Steam API calls
+            const html = await RequestData.getHttp(User.profileUrl);
+            const dummyPage = HTMLParser.htmlToDOM(html);
+            const config = dummyPage.getElementById("application_config").dataset.loyaltystore;
+            this.userToken = JSON.parse(config).webapi_token;
+
+            try {
+                const [equipped, owned] = await Promise.all([
+                    RequestData.getJson(`https://api.steampowered.com/IPlayerService/GetProfileItemsEquipped/v1?access_token=${this.userToken}&steamid=${User.steamId}`, {"credentials": "omit"}),
+                    RequestData.getJson(`https://api.steampowered.com/IPlayerService/GetProfileItemsOwned/v1?access_token=${this.userToken}`, {"credentials": "omit"})
+                ]);
+
+                this.profileBgItemId = equipped.response.profile_background.communityitemid; // undefined if background unset
+                this.profileBgsOwned = owned.response.profile_backgrounds;
+            } catch (err) {
+                console.error(err);
+                return;
+            }
+        }
+
+        // Loop through owned backgrounds to find the communityitemid for selected background
+        if (!thisItem.dataset.communityitemid) {
+
+            const m = viewFullBtn.href.match(/items\/\d+\/.+\.jpg/);
```

Comment:
Are preview images always in `.jpg`?

Reply:
Perhaps its better to be more flexible with the extension.

Reply:
Then maybe remove the extension part from the regex?